built grading_system

My-Boy001 opened the pull request.

nitin272 committed:
built grading_system

My-Boy001 commented:
@CodiumAI-Agent  /review

CodiumAI-Agent commented:
## PR Reviewer Guide 🔍

Here are some key observations to aid the review process:

<table>
<tr><td>⏱️&nbsp;<strong>Estimated effort to review</strong>: 2 🔵🔵⚪⚪⚪</td></tr>
<tr><td>🧪&nbsp;<strong>No relevant tests</strong></td></tr>
<tr><td>🔒&nbsp;<strong>No security concerns identified</strong></td></tr>
<tr><td>⚡&nbsp;<strong>Recommended focus areas for review</strong><br><br>

<details><summary><a href='https://github.com/My-Boy001/Ruby_On_Rails/pull/2/files#diff-dfe931ae89a11b1af08da807ea10bec29410ed5908aca0c88b651658fe4436dbR2-R2'><strong>Input Validation</strong></a>

The script converts any non-numeric input to 0 and allows out-of-range scores, which may lead to incorrect grading. Consider validating that `score` is numeric and within 0–100 and providing clear error messages for invalid input.
</summary>

```ruby
score = gets.chomp.to_i  
```

</details>

<details><summary><a href='https://github.com/My-Boy001/Ruby_On_Rails/pull/2/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R48-R48'><strong>Inconsistent Project Name</strong></a>

The setup instructions reference `Ruby_On_Rails` as the directory name, but the project is `grade_system`. This mismatch could confuse users following the guide. Update the directory references to match the actual project name.
</summary>

```markdown
cd Ruby_On_Rails # Replace 'Ruby_On_Rails' with your repository's directory name
```

</details>

</td></tr>
</table>

My-Boy001 commented:
### Input Handling:

I started by prompting the user to enter their score using puts. Then, I captured the input with gets.chomp to get a clean string without newline characters and converted it to an integer with .to_i. This allows numeric comparisons later.

> 1. Conditional Logic for Grading:
> I used an if-elsif-else chain to assign grades based on the input score. The conditions check from highest grade (A) down to lowest (F), comparing the score against thresholds (90, 80, 70, 60).
> 
> 2. User Feedback with Emojis:
> For each grade output, I added a relevant emoji to make the feedback more engaging and visually clear (e.g., 🎓 for A, ❌ for F).
> 
> 3. Edge Case Handling:
> The else condition catches any scores below 60, assigning an F grade, ensuring all possible inputs are covered.

CodiumAI-Agent commented:
**[Persistent review](https://github.com/My-Boy001/Ruby_On_Rails/pull/2#issuecomment-2899783316)** updated to latest commit https://github.com/My-Boy001/Ruby_On_Rails/commit/c5aed4c6dd2854596320a40a0eb751880cd98e86
